src/Microsoft.Azure.WebJobs.Host/Extensions/ExceptionExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace System
+{
+    internal static class ExceptionExtensions
+    {
+        public static bool IsFatal(this Exception exception)
+        {
+            while (exception != null)
+            {
+                if ((exception as OutOfMemoryException != null && exception as InsufficientMemoryException == null) ||
```

Comment:
InsufficientMemoryException is an OutOfMemoryException

So we want to throw on OOM but not IME.

---
In reply to: [76674693](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r76674693) [](ancestors = 76674693)